drivers/bluetooth/hci/spi.c

```diff
@@ -302,36 +287,44 @@ static void bt_spi_rx_thread(void)
 	uint8_t size = 0U;
 	int ret;
 	int len;
+	uint8_t success, is_irq_pin_high;
 
 	(void)memset(&txmsg, 0xFF, SPI_MAX_MSG_LEN);
-
+	/* Enable the interrupt line */
+	gpio_pin_interrupt_configure_dt(&irq_gpio, GPIO_INT_EDGE_TO_ACTIVE);
 	while (true) {
-		/* Enable the interrupt line */
-		gpio_init_callback(&gpio_cb, bt_to_active_isr, BIT(irq_gpio.pin));
-		gpio_pin_interrupt_configure_dt(&irq_gpio, GPIO_INT_LEVEL_ACTIVE);
 
 		/* Wait for interrupt pin to be active */
 		k_sem_take(&sem_request, K_FOREVER);
 
-		/* Wait for SPI bus to be available */
-		k_sem_take(&sem_busy, K_FOREVER);
-
 		LOG_DBG("");
 
-		do {
+		/* On BlueNRG-MS, host is expected to read */
```

Comment:
This is a common driver. Please do not refer to specific implementations here, or then encapsulate them inside a Kconfig option that does not refer to the particular SoC

Reply:
Can't fault @HoZHel for this, they've just moved this comment from elsewhere in the file :)
Personally I almost feel like the BlueNRG driver and the Zephyr driver should be separate implementations, its annoying to make the driver work for both.